### PR TITLE
Hybridcompute - removed reconnect api

### PR DESCRIPTION
**Pull request by jianyunt**
This RP is primarily removing reconnect api for security considerations.

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [x] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2020-07-30-preview/HybridCompute.json [hunk 1]
```diff
@@ -0,0 +1,1211 @@
...
+          "readOnly": true,
+          "type": "string",
+          "description": "The Operating System running on the hybrid machine."
+        },
+        "osVersion": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The version of Operating System running on the hybrid machine."
+        },
+        "vmUuid": {
```
Comment (pilor) [inline specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2020-07-30-preview/HybridCompute.json:819]: if readonly please mark as readonly

File: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2020-07-30-preview/HybridCompute.json [hunk 2]
```diff
@@ -0,0 +1,1211 @@
...
+          "List Machines by resource group": {
+            "$ref": "./examples/Machines_ListByResourceGroup.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.HybridCompute/machines": {
```
Comment (pilor) [inline specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2020-07-30-preview/HybridCompute.json:309]: What response will users get when they call /reconnect in an older api-version?

**Review by pilor — CHANGES_REQUESTED** [review]

File: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2020-07-30-preview/HybridCompute.json (hunk 2 above)
Comment (jianyunt) [reply specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2020-07-30-preview/HybridCompute.json:309]: The response from ARM of reconnect is the same as connect. For example,
{
  "parameters": {
    "subscriptionId": "{subscriptionId}",
    "resourceGroupName": "myResourceGroup",
    "name": "myMachine",
    "api-version": "2019-12-12",
    "parameters": {
      "properties": {
        "vmId": "b7a098cc-b0b8-46e8-a205-62f301a62a8f",
        "clientPublicKey": "string"
      }
    }
  },
  "responses": {
    "200": {
      "body": {
        "id": "/subscriptions/{subscriptionId}/resourceGroups/myResourceGroup/providers/Microsoft.HybridCompute/machines/myMachine",
        "name": "myMachine",
        "location": "eastus2euap",
        "tags": null,
        "identity": {
          "type": "SystemAssigned",
          "principalId": "string",
          "tenantId": "string"
        },
        "type": "Microsoft.HybridCompute/machines",
        "properties": {
          "provisioningState": "Succeeded",
          "agentVersion": null,
          "status": null,
          "lastStatusChange": null,
          "errorDetails": null,
          "displayName": null,
          "machineFqdn": null,
          "osName": null,
          "osVersion": null,
          "osProfile": {
            "computerName": null
          },
          "vmId": "b7a098cc-b0b8-46e8-a205-62f301a62a8f",
          "locationData": {
            "name": "Redmond"
          },

        }
      }
    }
  }
}

File: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2020-07-30-preview/HybridCompute.json (hunk 1 above)
Comment (jianyunt) [reply specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2020-07-30-preview/HybridCompute.json:819]: added. thanks!

File: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2020-07-30-preview/HybridCompute.json (hunk 2 above)
Comment (jianyunt) [reply specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2020-07-30-preview/HybridCompute.json:309]: no error with reconnect if using old api for now. we will either update our RP code to respond "Deprecated".  Or simply remove reconnect from our RP roledefinition.
However we have removed the reconnect from our agent code already which is a downloadable package to be run on machines in the August release.

**Review by pilor — APPROVED** [review]

**Pull request merged**